public/app/plugins/datasource/grafana-azure-monitor-datasource/components/AnalyticsConfig.tsx

```diff
@@ -140,28 +143,39 @@ export const AnalyticsConfig: FunctionComponent<Props> = (props: Props) => {
       <h3 className="page-heading">Azure Monitor Logs</h3>
       {credentialsEnabled && (
         <>
-          <Switch
-            label="Same details as Azure Monitor API"
-            checked={!logAnalyticsCredentials}
-            onChange={onLogAnalyticsSameAsChange}
-            {...tooltipAttribute}
-          />
+          {showLogAnalyticsCreds && (
```

Comment:
The variable `credentialsEnabled` above serves the same purpose - to hide the credentials section when not needed., You can change condition of that variaable instead.